src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -280,39 +282,34 @@ public bool IsSolutionAvailable
         {
             get
             {
-                return NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+                if (!IsSolutionOpen)
                 {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
We may need to run this api on ui thread or make sure all the calls inside this method are moving to UI thread as when applicable. For starter, DoesSolutionRequireAnInitialSaveAs tries to retrieve vs solution properties but i doubt it explicitly go to UI thread...

Besides I doubt it will make much of the difference taking this api off the UI thread because anyways in most cases, it will still be doing context switching to UI thread anyways.